android/BOINC/app/src/main/java/edu/berkeley/boinc/ProjectsFragment.java

```diff
@@ -185,7 +185,7 @@ private void updateData(ArrayList<Project> latestRpcProjectsList, AcctMgrInfo ac
 			for(int x = 0; x < data.size(); x++) {
 				if(rpcResult.master_url.equals(data.get(x).id)) {
 					index = x;
-					continue;
+					break;
```

Comment:
Should be ok as well as finding a single match seems to be the intention here.

Reply:
Same as above. Need more info

Reply:
I disagree. The comment above says, "Check whether this Result is new". If we pass the `index` a value, the `if (index == null)` is already false, so we don't need to find more `X`s.

Reply:
@adamradocz, I agree with you but the open question for me now it why this function is implemented this way. Maybe the current behavior is incorrect at whole but changing the logic without full understanding of the situation in worse than just leaving it as is for a while.
Please understand me correctly, I also want to make this product better, as all of us I believe.
And thank you for your hard job!

Reply:
I think the whole codebase shows that the original author had no clear understanding of what `continue` means. So we have to take that into account while trying to determine the **originally intended** functionality.

Reply:
I made the changes that @AenBleidd proposed.

Reply:
Same as above: removing `continue` doesn't change the semantics and the proposed solution is visible, yet disabled until further clarification.